### ISSUE TITLE
Can't use `array<string, string>` as a property doc

### ISSUE THREAD

**Issue by Einenlum** [nelmio/NelmioApiDocBundle#1816]
Hello. I'm using the version 3.8.2.

I found a very strange bug. Here is the model I'm using:

```php
<?php

declare(strict_types=1);

namespace Domain\Model\Mail;

class SendMailDTO
{
    private array $params = [];

    /** @return array<string, string> */
    public function getParams(): array
    {
        return $this->params;
    }

    /** @param array<string, string> $params */
    public function setParams(array $params): void
    {
        $this->params = $params;
    }
```

And the Symfony controller:

```php
<?php

declare(strict_types=1);

namespace App\Controller\V1\Mail;

use Domain\Model\Mail\SendMailDTO;
use Infrastructure\Symfony\ParamConverter\SymfonySerializerConverter;
use Nelmio\ApiDocBundle\Annotation as Nelmio;
use Sensio\Bundle\FrameworkExtraBundle\Configuration\ParamConverter;
use Swagger\Annotations as SWG;
use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
use Symfony\Component\HttpFoundation\JsonResponse;
use Symfony\Component\HttpFoundation\Request;
use Symfony\Component\HttpFoundation\Response;
use Symfony\Component\Routing\Annotation\Route;
use Symfony\Component\Validator\ConstraintViolationListInterface;

class SendMailController extends AbstractController
{
    /**
     * @SWG\Post(
     *     description="Some description",
     *     @SWG\Schema(ref=@Nelmio\Model(type=SendMailDTO::class))
     * )
     *
     * @SWG\Response(
     *     response=Response::HTTP_OK,
     *     description="The mail was sent"
     * )
     * @SWG\Response(
     *     response=Response::HTTP_BAD_REQUEST,
     *     description="Invalid payload"
     * )
     * @Route("/some-route", name="some_name", methods={Request::METHOD_POST})
     * @ParamConverter("dto", converter=SymfonySerializerConverter::class)
     */
    public function __invoke(string $customerReference, SendMailDTO $dto, ConstraintViolationListInterface $validationErrors): JsonResponse
    {
        // ...
        return new JsonResponse(['message' => 'MAIL_SENT'], Response::HTTP_OK);
    }
}
```

I get this error message: `Schema of type \"rray<string,string>\" can't be generated, no describer supports it.`, coming from [here](https://github.com/nelmio/NelmioApiDocBundle/blob/v3.8.2/Model/ModelRegistry.php#L96).

If I change my DTO to use `aarray` instead of `array`, it then works :eyes: : 

```php
<?php

declare(strict_types=1);

namespace Domain\Model\Mail;

class SendMailDTO
{
    private array $params = [];

    /** @return aarray<string, string> */
    public function getParams(): array
    {
        return $this->params;
    }

    /** @param aarray<string, string> $params */
    public function setParams(array $params): void
    {
        $this->params = $params;
    }
}
```

Is there some bug regarding how doctypes are parsed?

**Comment by GuilhemN** [comment]
It's an issue in symfony/property-info, and it has been fixed in version 5.2 (see https://github.com/symfony/symfony/commit/64f7bd78323dd1a3d544a2b93674642d25593ede).

There is not much we can do here, you can either upgrade this symfony component or use an explicit version like `@SWG\Parameter(type="array", @SWG\Items(type="string"))`.

**Comment by Einenlum** [comment]
Oh good to know, thx @GuilhemN !
